modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/BasicIndexTest.java

```diff
@@ -229,6 +257,44 @@ public void testNoIndexesNoPersistence() throws Exception {
         }
     }
 
+    /** */
+    @Test
+    public void testInStatementUsesIndex() throws Exception {
+        H2TreeIndex.h2TreeFactory = TestH2Tree::new;
+
+        inlineSize = 30;
+
+        IgniteEx ig0 = startGrid();
+        GridQueryProcessor qryProc = ig0.context().query();
+
+        Function<String, FieldsQueryCursor<List<?>>> execSql =
+            sql0 -> qryProc.querySqlFields(new SqlFieldsQuery(sql0), true);
+
+        execSql.apply("CREATE TABLE PUBLIC.TEST_TABLE (f1 int, f2 int, f3 int, f4 int, PRIMARY KEY (f1, f2, f3))");
+        execSql.apply("CREATE INDEX PK_IDX ON PUBLIC.TEST_TABLE (f1, f2, f3)");
+
+        String sqlInsert = "INSERT INTO PUBLIC.TEST_TABLE (f1, f2, f3, f4) values (%d, %d, %d, 2)";
+
+        for (int i = 0; i < 10; ++i)
+            for (int j = 0; j < 10; ++j)
+                for (int k = 0; k < 10; ++k)
+                    execSql.apply(String.format(sqlInsert, i, j, k));
+
+        Consumer<String> execAssert = sql -> {
+            scanCntr.set(0);
+
+            FieldsQueryCursor<List<?>> res =
+                execSql.apply(sql);
+
+            assertEquals(4, res.getAll().size());
+            assertEquals(4, scanCntr.get());
+        };
+
+        execAssert.accept("SELECT * FROM PUBLIC.TEST_TABLE WHERE f1 = 8 and f2 in (1, 5, 3, 7) and f3 = 5");
+        execAssert.accept("SELECT * FROM PUBLIC.TEST_TABLE WHERE f1 in (1, 5, 3, 7) and f2 = 8 and f3 = 5");
+        execAssert.accept("SELECT * FROM PUBLIC.TEST_TABLE WHERE f1 = 8 and f2 = 5 and f3 in (1, 5, 3, 7)");
```

Comment:
plz append additionally tests for :
`"SELECT * FROM PUBLIC.TEST_TABLE WHERE f1 = 8 and f2 in (1) and f3 in (1, 5, 3, 7)"`
`"SELECT * FROM PUBLIC.TEST_TABLE WHERE f1 = 8 and f2 in (1, 5) and f3 in (1, 5, 3, 7)"
"SELECT * FROM PUBLIC.TEST_TABLE WHERE f1 in (1, 5) and f2 = 5 and f3 in (1, 5, 3, 7)"`